### DISCUSSION TITLE
where does the $secret.notation in the values.yaml come from?

### DISCUSSION THREAD

**Discussion by f2calv** [argoproj/argo-helm#2223] [Q&A] [answered]
I've seen this secret notation in the argocd values.yaml, $dex.github.clientSecret or $oidc.azuread.clientSecret and I wondered if this is a custom ArgoCD thing or a general Helm method of pointing to secrets? Does anyone have any idea as I was unable to find any relevant documentation...

![image](https://github.com/argoproj/argo-helm/assets/16097639/46bf5ded-708b-403e-9375-6b05118c7195)

**Comment by yu-croco** [comment, marked as answer]
Hi @f2calv ,
I think this is from [upstream's doc](https://argo-cd.readthedocs.io/en/stable/operator-manual/user-management/#2-configure-argo-cd-for-sso) and it is just an example...? 🤔 
Since this is about Argo CD's feature, so I think you can get more help in [upstream](https://github.com/argoproj/argo-cd/discussions).

**Reply by pdrastil** [reply]
As @yu-croco mentioned Argo CD is watching the `argocd-secret` for keys that should be substituted with real value. So to answer your question the `$<key>` notation is specific only to Argo CD.
